src/Tasks/SystemState.cs

```diff
@@ -607,20 +600,8 @@ private string[] GetDirectories(string path, string pattern)
         /// <returns>True if the file exists.</returns>
         private bool FileExists(string path)
         {
-            /////////////////////////////////////////////////////////////////////////////////////////////
-            // FIRST -- Look in the primary cache for this path.
-            /////////////////////////////////////////////////////////////////////////////////////////////
-            object flag = instanceLocalFileExists[path];
-            if (flag != null)
-            {
-                return (bool)flag;
-            }
-
-            /////////////////////////////////////////////////////////////////////////////////////////////
-            // SECOND -- fall back to plain old File.Exists and cache the result.
-            /////////////////////////////////////////////////////////////////////////////////////////////
-            bool exists = fileExists(path);
-            instanceLocalFileExists[path] = exists;
+            DateTime lastModified = GetFileState(path).LastModified;
+            bool exists = lastModified != DateTime.MinValue && lastModified != DateTime.FromFileTimeUtc(0);
```

Comment:
Ideally only one of these would be necessary, but I've noticed MSBuild has mixed usage on whether a non-existent file is represented by ```DateTime.MinValue``` or ```DateTime.FromFIleTimeUtc(0)```, which is what .NET uses. And since ```GetLastWriteTime``` here is a delegate, can't really be sure of which one.

Reply:
Second concern: since nothing else in the class checks for an invalid file time, I also can't tell if ```GetFileState()``` assumes that the path is going to be going to be valid, or if it's assuming the delegates will handle an invalid file. It would easy to workaround (keep ```instanceLocalFileExists``` and only cache ```FileState``` for existing files) but it's not clear.

Reply:
Since this should be moving out of proc I think we can start enforcing a standard, but this seems good for now. Can you file a followup work item and link it here in a `// TODO`?